contrib/pyln-testing/pyln/testing/utils.py

```diff
@@ -200,24 +200,29 @@ def __init__(self, outputDir, verbose=True):
         # pass it to the log matcher and not print it to stdout).
         self.log_filter = lambda line: False
 
-    def start(self, stdin=None, stdout_redir=True):
+    def start(self, stdin=None, stdout_redir=True, stderr_redir=True):
         """Start the underlying process and start monitoring it. If
         stdout_redir is false, you have to make sure logs go into
         outputDir/log
 
         """
         logging.debug("Starting '%s'", " ".join(self.cmd_line))
         if stdout_redir:
-            self.proc = subprocess.Popen(self.cmd_line,
-                                         stdin=stdin,
-                                         stdout=self.stdout_write,
-                                         stderr=self.stderr_write,
-                                         env=self.env)
+            stdout = self.stdout_write
         else:
-            self.proc = subprocess.Popen(self.cmd_line,
-                                         stdin=stdin,
-                                         stderr=self.stderr_write,
-                                         env=self.env)
+            stdout = None
+        if stderr_redir:
+            stderr = self.stderr_write
+            self.stderr_redir = True
+        else:
+            stderr = None
```

Comment:
Any reason we still open the `self.stderr_write` and `self.stderr_read` if we never intend to use them?